src/victory-legend/victory-legend.js

```diff
@@ -0,0 +1,167 @@
+      labels: PropTypes.object
+    })
+  };
+
+  static defaultProps = {
+    x: 0,
+    y: 0,
+    height: 100,
+    width: 150,
+    orientation: "vertical",
+    columnItemSpacing: 10,
+    rowItemSpacing: 8,
+    dataComponent: <Point/>,
+    labelComponent: <VictoryLabel/>,
+    containerComponent: <VictoryContainer/>,
+    groupComponent: <g/>,
+    standalone: true,
+    style: {}
+  };
+
+  getLabelStyles(props) {
+    return merge({}, fallbackProps.style.labels, props.label);
+  }
+
+  getLegendStyle(props) {
+    return {
+      style: defaults({}, defaultLegendStyle, props.style)
+    };
+  }
+
+  renderContainer(props, group) {
+    return React.cloneElement(
+      props.containerComponent,
+      this.getLegendStyle(props),
+      group
+    );
+  }
+
+  renderGroup(children) {
+    return React.cloneElement(
+      this.props.groupComponent,
+      { role: "presentation" },
+      children
+    );
+  }
+
+  getSymbolProps(series) {
+    return assign({}, fallbackProps.style.symbol, this.props.style.symbol, series.symbol);
+  }
+
+  render() {
+    let yPadding = 0;
+    let xPadding = 0;
+    const props = this.props;
+    const itemXPosition = [];
+    const dataComponents = [];
+    const labelComponents = [];
+    const { dataComponent, labelComponent } = props;
+    const data = isEmpty(props.data) ? defaultLegendData : props.data;
+
+    data.forEach((series, i) => {
+      const font = this.getLabelStyles(series);
+      const symbolSize = font.fontSize;
+      const hPadding = symbolSize * 0.87;
+      const yPos = props.y + i * yPadding;
+      const labelSymbolPadding = symbolSize * 1.5 + props.columnItemSpacing;
+
+      if (props.orientation === "horizontal") {
+        itemXPosition[i] = xPadding;
+        xPadding += labelSymbolPadding + Textsize.approximateTextSize(series.name, font).width;
+      } else {
+        yPadding = symbolSize + props.rowItemSpacing;
+      }
+
+      const xLabelPos = (props.orientation !== "horizontal" ?
+        props.x : itemXPosition[i]) + hPadding;
+      const xSymbolPos = props.orientation !== "horizontal" ?
+        props.x : itemXPosition[i];
+
+      const symbolProps = assign({
+        key: `symbol-${i}`
+      }, {
+        x: xSymbolPos,
+        y: yPos,
+        size: series.symbol ? series.symbol.size || symbolSize / 2.5 : symbolSize / 2.5,
+        symbol: this.getSymbolProps(series).type
+      }, this.getSymbolProps(series));
+
+      const labelProps = assign({
+        key: `label-${i}`
+      }, {
+        x: xLabelPos,
+        y: yPos,
+        style: font,
+        text: series.name
+      }, assign({}, fallbackProps.style.labels, props.style.labels, series.label));
+
+      dataComponents[i] = React.cloneElement(
+        dataComponent, assign({}, symbolProps)
+      );
+      labelComponents[i] = React.cloneElement(
+        labelComponent, assign({}, labelProps)
+      );
+    });
+
+    const group = this.renderGroup([...dataComponents, ...labelComponents]);
```

Comment:
I would prefer if calculating all the dataComponents and labelComponents was moved outside of the render method like:

```
renderLegend(props) {
  ...
  return [...dataComponents, ...labelComponents];
}

render() {
   // do all your style / theme / fallback prop merging here
   const group = this.renderGroup(this.renderLegend(reconciledProps));
   return props.standalone ? this.renderContainer(props, group) : group;
}